src/core_plugins/console/server/__tests__/elasticsearch_proxy_config.js

```diff
@@ -118,22 +122,50 @@ describe('plugins/console', function () {
         expect(agent.options.ca).to.contain('test ca certificate\n');
       });
 
-      it(`doesn't set cert and key when certificate and key paths are specified`, function () {
-        setElasticsearchConfig('ssl.certificate', __dirname + '/fixtures/cert.crt');
-        setElasticsearchConfig('ssl.key', __dirname + '/fixtures/cert.key');
-
-        const { agent } = getElasticsearchProxyConfig(server);
-        expect(agent.options.cert).to.be(undefined);
-        expect(agent.options.key).to.be(undefined);
+      describe('when alwaysPresentCertificate is false', () => {
+        it(`doesn't set cert and key when certificate and key paths are specified`, function () {
+          setElasticsearchConfig('ssl.alwaysPresentCertificate', false);
+          setElasticsearchConfig('ssl.certificate', __dirname + '/fixtures/cert.crt');
+          setElasticsearchConfig('ssl.key', __dirname + '/fixtures/cert.key');
+
+          const { agent } = getElasticsearchProxyConfig(server);
+          expect(agent.options.cert).to.be(undefined);
+          expect(agent.options.key).to.be(undefined);
+        });
+
+        it(`doesn't set passphrase when certificate, key and keyPassphrase are specified`, function () {
+          setElasticsearchConfig('ssl.alwaysPresentCertificate', false);
+          setElasticsearchConfig('ssl.certificate', __dirname + '/fixtures/cert.crt');
+          setElasticsearchConfig('ssl.key', __dirname + '/fixtures/cert.key');
+          setElasticsearchConfig('ssl.keyPassphrase', 'secret');
+
+          const { agent } = getElasticsearchProxyConfig(server);
+          expect(agent.options.passphrase).to.be(undefined);
+        });
       });
 
-      it(`doesn't set passphrase when certificate, key and keyPassphrase are specified`, function () {
-        setElasticsearchConfig('ssl.certificate', __dirname + '/fixtures/cert.crt');
-        setElasticsearchConfig('ssl.key', __dirname + '/fixtures/cert.key');
-        setElasticsearchConfig('ssl.keyPassphrase', 'secret');
-
-        const { agent } = getElasticsearchProxyConfig(server);
-        expect(agent.options.passphrase).to.be(undefined);
+      describe('when alwaysPresentCertificate is true', () => {
```

Comment:
What do you think about adding a test that ensures that the certificate isn't set if the key isn't also set? (and/or vice versa?) Basically, we require both the cert and the key to be specified before we'll set either of them.